lib/webmachine/adapters/rack.rb

```diff
@@ -95,6 +92,14 @@ def call(env)
         rack_res.finish
       end
 
+      private
+      def build_webmachine_request(rack_req, headers)
+        Webmachine::Request.new(rack_req.request_method,
+                                rack_req.url,
+                                headers,
+                                RequestBody.new(rack_req))
+      end
+
```

Comment:
Yup. I like that approach. Feels a lot cleaner and easier to follow. Will implement that.